client/dashboard/me/monetize-subscriptions/monetize-subscription.tsx

```diff
@@ -0,0 +1,307 @@
+	subscription: MonetizeSubscription;
+} ) {
+	const title = isAutoRenewing ? __( 'Disable auto-renew' ) : __( 'Enable auto-renew' );
+	return (
+		<ActionList.ActionItem
+			title={ title }
+			description={ title }
+			actions={
+				<Button
+					variant="secondary"
+					size="compact"
+					disabled={ isUpdating }
+					onClick={ isAutoRenewing ? disableAutoRenew : enableAutoRenew }
+				>
+					{ title }
+				</Button>
+			}
+		/>
+	);
+}
+
+function StopSubscriptionButton( {
+	stopSubscription,
+	isProduct,
+	subscription,
+}: {
+	stopSubscription: () => void;
+	isProduct: boolean;
+	subscription: MonetizeSubscription;
+} ) {
+	const title = isProduct
+		? // eslint-disable-next-line @wordpress/i18n-translator-comments
+		  sprintf( __( 'Remove %s product' ), subscription.title )
+		: // eslint-disable-next-line @wordpress/i18n-translator-comments
+		  sprintf( __( 'Stop %s subscription' ), subscription.title );
+	return (
+		<ActionList.ActionItem
+			title={ title }
+			description={ title }
+			actions={
+				<Button
+					variant="secondary"
+					size="compact"
+					onClick={ () => {
+						stopSubscription();
+					} }
+				>
+					{ title }
+				</Button>
+			}
+		/>
+	);
+}
+
+const BackButton = () => {
+	const router = useRouter();
+
+	return (
+		<Button
+			className="dashboard-page-header__back-button"
+			icon={ isRTL() ? chevronRight : chevronLeft }
+			onClick={ () => {
+				router.navigate( {
+					to: getMonetizeSubscriptionsUrl(),
+				} );
+			} }
+		>
+			{ getMonetizeSubscriptionsPageTitle() }
+		</Button>
+	);
+};
+
+export default function MonetizeSubscription() {
+	const params = monetizeSubscriptionRoute.useParams();
+	const subscriptionId: string = params.subscriptionId;
+
+	const { data: subscription } = useQuery( monetizeSubscriptionQuery( subscriptionId ) );
+
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+
+	const {
+		mutate: stopSubscription,
+		isError: stoppingStatusError,
+		isPending: isStoppingSubscription,
+		isSuccess: stoppingStatusSuccess,
+	} = useMutation( monetizeSubscriptionStop( subscriptionId ) );
+
+	const {
+		mutate: disableAutoRenew,
+		isError: disableAutoRenewError,
+		isPending: isDisablingAutoRenew,
+	} = useMutation( monetizeSubscriptionDisableAutoRenew( subscriptionId ) );
+
+	const {
+		mutate: enableAutoRenew,
+		isError: enableAutoRenewError,
+		isPending: isEnablingAutoRenew,
+	} = useMutation( monetizeSubscriptionResumeAutoRenew( subscriptionId ) );
+
+	const isRenewable = subscription && ( subscription.renew_interval || subscription.is_renewable ); // can remove renew_interval once backend is deployed
+	const isAutoRenewing = isRenewable && subscription.renew_interval;
+	const isProduct = !! subscription && ! isRenewable;
+	const isDisabledAutorenewing = isRenewable && ! subscription.renew_interval;
+	const isUpdating = isEnablingAutoRenew || isDisablingAutoRenew;
+	const navigate = useNavigate();
+	useEffect( () => {
```

Comment:
I haven't looked into it deeply, but using `useEffect` for error handling seems like a code smell. We shouldn't need to use a `useEffect` to listen for the value of `stoppingStatusError` to change, when we can listen to `onError` of the `monetizeSubscriptionStop` mutation e.g.

```ts
stopSubscription( null, {
  onError: () => {
    // error notice
  }
} );
```

Reply:
Yes, I agree. I merely reuse the original code. I should not have.